src/soilId/soilIdSlice.ts

```diff
@@ -163,8 +166,9 @@ const soilIdSlice = createSlice({
     });
 
     builder.addCase(fetchSoilIdMatches.fulfilled, (state, action) => {
-      state.soilIdStatus = 'ready';
-      state.soilIdData = action.payload;
+      const { failureReason, ...payload } = action.payload;
+      state.soilIdStatus = failureReason ?? 'ready';
```

Comment:
Question: Hmm, I'm seeing that on line 165 we will set the soilIdStatus to 'error' in the rejected case, wondering how that fits in with this change. Do we expect to only be in the rejected case when our backend can't respond? 

I don't have a clear take on if there may be a clearer way to organize; I think I'm kinda just like "oh there's this one type of error that exists one level up from the other types of errors, does this make sense?"

Reply:
yeah, so the 3 different `fulfilled`, `rejected`, and `pending` cases are the cases for what happened to the request itself. we would get into a `rejected` state and set `soilIdStatus` to `error` if the request fails (e.g. network conditions, malformed graphql query, etc), in which case we never actually hit the soil ID algorithm.

the code you quoted here is for the case when the _request_ returns successfully, but it could be returning with news that the algorithm failed

arguably (i might argue) it would make more sense to have the request itself also fail in that situation, but that would require refactoring in the vein of "our backend error handling is inconsistent and confusing" which i didn't want to get into right now